src/parachain/issue.ts

```diff
@@ -231,18 +240,28 @@ export class DefaultIssueAPI extends DefaultTransactionAPI implements IssueAPI {
             const remainder = amount.sub(successfulSum);
             if (remainder.isZero() || retries === 0) return result;
             else {
-                return (await this.request(remainder, vaultId, atomic, retries - 1, availableVaults)).concat(result);
+                return (
+                    await this.request(
+                        remainder,
+                        vaultAccountId,
+                        collateralCurrencyIdLiteral,
+                        atomic,
+                        retries - 1,
+                        availableVaults
+                    )
+                ).concat(result);
             }
         } catch (e) {
             return Promise.reject(e);
         }
     }
 
     async craftRequestTx(
-        vaultId: AccountId,
+        vaultId: InterbtcPrimitivesVaultId,
         amount: MonetaryAmount<WrappedCurrency, BitcoinUnit>
     ): Promise<SubmittableExtrinsic<"promise">> {
-        const vault = await this.vaultsAPI.get(vaultId);
+        const collateralId = currencyIdToLiteral(vaultId.currencies.collateral);
+        const vault = await this.vaultsAPI.get(vaultId.accountId, collateralId);
         let griefingCollateral = await this.getGriefingCollateral(amount, vault.collateralCurrency);
         // add() here is a hacky workaround for rounding errors
         const oneHundred = newMonetaryAmount(100, vault.collateralCurrency);
```

Comment:
Why is the griefing collateral in `Satoshi`? The griefing collateral is in DOT/KSM